workshops/management/commands/check_for_workshop_websites_updates.py

```diff
@@ -0,0 +1,244 @@
+from collections.abc import Iterable
+import datetime
+from functools import partial
+import json
+
+from django.core.management.base import BaseCommand
+from github import Github
+import requests
+from rest_framework.utils.encoders import JSONEncoder
+
+from workshops.models import Event
+from workshops.util import (
+    fetch_event_tags,
+    parse_tags_from_event_website,
+    WrongWorkshopURL,
+)
+
+
+def datetime_match(string):
+    """Convert string date/datetime/time to date/datetime/time."""
+    try:
+        # try parsing date
+        v = datetime.datetime.strptime(string, '%Y-%m-%d')
+        return v.date()
+    except ValueError:
+        pass
+
+    try:
+        # try parsing datetime (no microseconds, timezone unaware)
+        return datetime.datetime.strptime(string, '%Y-%m-%dT%H:%M:%S')
+    except ValueError:
+        pass
+
+    try:
+        # try parsing datetime (w/ microseconds, timezone unaware)
+        return datetime.datetime.strptime(string,
+                                          '%Y-%m-%dT%H:%M:%S.%f')
+    except ValueError:
+        pass
+
+    # TODO: Implement timezone-aware datetime parsing (currently
+    #       not available because datetime.datetime.strptime
+    #       doesn't support "+HH:MM" format [only "+HHMM"]; nor
+    #       does it support "Z" at the end)
+
+    try:
+        # try parsing time (no microseconds, timezone unaware)
+        v = datetime.datetime.strptime(string, '%H:%M:%S')
+        return v.time()
+    except ValueError:
+        pass
+
+    try:
+        # try parsing time (w/ microseconds, timezone unaware)
+        v = datetime.datetime.strptime(string, '%H:%M:%S.%f')
+        return v.time()
+    except ValueError:
+        pass
+
+    return string
+
+
+def datetime_decode(obj):
+    """Recursively call for each iterable, and try to decode each string."""
```

Comment:
Why all this complexity? Can we not simply poke the page author if parsing fails?